src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -421,18 +517,20 @@ end
 
 # Convert primitive to conservative variables
 @inline function Trixi.prim2cons(prim, equations::ShallowWaterMultiLayerEquations1D)
-    H = prim[1:nlayers(equations)]
-    v = prim[(nlayers(equations) + 1):(2 * nlayers(equations))]
+    # To extract the interface height and velocity we reuse the waterheight and momentum functions 
+    # from the conservative variables.
+    H = waterheight(prim, equations)    # For primitive variables this extracts the interface height
+    v = momentum(prim, equations)       # For primitive variables this extracts the velocity
```

Comment:
but isn't there already a `velocity` function?

Reply:
Yes, but they are written for conservative variables and here we have primitive variables as input. 
The `velocity` function would return `v/H` for primitive variables, therefore I am reusing the `momentum` to extract the respective elements from the solution vector.

Reply:
Aha, gotcha. Then this is a convenient way to grab the values you need.